### PR TITLE
Add note about mount propagation on systemd

### LINKED ISSUE #19625: Shared subtrees not working under Debian
I'm having issues getting the new `--volume /mnt/shared:/shared:shared` feature of 1.10 working (from #17034). 

```bash
root@testhost:~# uname -a
Linux testhost 3.18.5-031805-generic #201501292218 SMP Fri Jan 30 03:19:17 UTC 2015 x86_64 GNU/Linux

root@testhost:~# docker info
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 9
Server Version: 1.10.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: host bridge null
Kernel Version: 3.18.5-031805-generic
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 1.899 GiB
Name: loxflix
ID: RNAT:QYNF:66KE:DSN5:NRPC:ITAA:6G3M:WMR2:JABB:CPGO:ZN7O:FAVU
Debug mode (server): true
 File Descriptors: 13
 Goroutines: 31
 System Time: 2016-01-24T08:45:00.139001319Z
 EventsListeners: 0
 Init SHA1: b84242d186971c8111d1f9de77b7c476bc049614
 Init Path: /usr/lib/docker/dockerinit
 Docker Root Dir: /var/lib/docker
Labels:
 provider=generic
Experimental: true

root@testhost:~# findmnt -o TARGET,PROPAGATION /mnt/testhost
TARGET       PROPAGATION
/mnt/shared shared

root@testhost:~# docker run --rm -it --volume /mnt/shared:/data:shared ubuntu:14.04
docker: Error response from daemon: Cannot start container 6859ea4fc2f23130da7e72f301c9a82528c6e78e406b6f68261d000410ad6960: Path /mnt/shared is mounted on /mnt/shared but it is not a shared mount..
```

I had this working under ubuntu. One of the key differences I noted was that `mount` indicated that the mount `/shared` had `-o bind` where as under debian, it doesn't seem to.

### PR DESCRIPTION
**Pull request by cpuguy83**
Closes #19625

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @thaJeztah @vdemeester for docs review

File: docs/reference/run.md [hunk 1]
```diff
@@ -1385,6 +1385,17 @@ Similarly the operator can set the **hostname** with `-h`.
...
+> **Note**:
+> When using systemd to manage the Docker daemon's start and stop, in the systemd
+> unit file there is an option to control mount propagation for the Docker daemon
+> itself, called `MountFlags`. The value of this setting may cause Docker to not
+> see mount propagation changes made on the mount point. For example, if this value
+> is `slave`, you may not be able to use the `shared` or `rshared` propagation on
+> a volume.
+
+
+
```
Comment (jfrazelle) [inline docs/reference/run.md]: why so many returns jeez its like you are running linux and used too much pressure :smiling_imp:

Comment (jfrazelle) [inline docs/reference/run.md]: btw you dont have to fix that im kidding

**Comment by jfrazelle** [comment]
LGTM

File: docs/reference/run.md (hunk 1 above)
Comment (cpuguy83) [inline docs/reference/run.md]: lol, no idea what happened.

**Comment by thaJeztah** [comment]
LGTM, we'll tweak later if needed. Thanks @cpuguy83!

**Comment by anusha-ragunathan** [comment]
LGTM

**Pull request merged**